Grid/Source/Entity.php

```diff
@@ -777,4 +781,24 @@ public function getTableAlias()
     {
         return $this->tableAlias;
     }
+
+    /**
+     * @param QueryBuilder $qb
+     * @return boolean
+     */
+    protected function checkIfQueryHasFetchJoin(QueryBuilder $qb)
+    {
+        $join = $qb->getDqlPart('join');
+        if (empty($join)) {
+            return false;
+        }
+
+        foreach ($join[$this->getTableAlias()] as $join) {
+            if ($join->getJoinType() === Join::INNER_JOIN) {
```

Comment:
Is there a reason you only check for inner joins here?

I tested this and it all works fine unless you have inner joins. In my case the QueryBuilder is built with only left joins, which then skips this check. 

The following would solve it;

```php
if ($join->getJoinType() === Join::INNER_JOIN || $join->getJoinType() === Join::LEFT_JOIN) {
    return true;
}
```